# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/sandbox_types.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/sandbox_types.h
--- a/security/sandbox/chromium/sandbox/win/src/sandbox_types.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/sandbox_types.h	Wed Jun 12 11:10:48 2019 +0100
@@ -113,6 +113,12 @@
   SBOX_ERROR_CANNOT_WRITE_INTERCEPTION_THUNK = 42,
   // Cannot find the base address of the new process.
   SBOX_ERROR_CANNOT_FIND_BASE_ADDRESS = 43,
+  // Cannot create the AppContainer profile.
+  SBOX_ERROR_CREATE_APPCONTAINER_PROFILE = 44,
+  // Cannot create the AppContainer as the main executable can't be accessed.
+  SBOX_ERROR_CREATE_APPCONTAINER_PROFILE_ACCESS_CHECK = 45,
+  // Cannot create the AppContainer as adding a capability failed.
+  SBOX_ERROR_CREATE_APPCONTAINER_PROFILE_CAPABILITY = 46,
   // Placeholder for last item of the enum.
   SBOX_ERROR_LAST
 };
@@ -131,9 +137,11 @@
   SBOX_FATAL_LAST
 };
 
+#if !defined(SANDBOX_FUZZ_TARGET)
 static_assert(SBOX_FATAL_MEMORY_EXCEEDED ==
                   base::win::kSandboxFatalMemoryExceeded,
               "Value for SBOX_FATAL_MEMORY_EXCEEDED must match base.");
+#endif  // !defined(SANDBOX_FUZZ_TARGET)
 
 class BrokerServices;
 class TargetServices;
@@ -152,7 +160,7 @@
 
 enum InterceptionType {
   INTERCEPTION_INVALID = 0,
-  INTERCEPTION_SERVICE_CALL,    // Trampoline of an NT native call
+  INTERCEPTION_SERVICE_CALL,  // Trampoline of an NT native call
   INTERCEPTION_EAT,
   INTERCEPTION_SIDESTEP,        // Preamble patch
   INTERCEPTION_SMART_SIDESTEP,  // Preamble patch but bypass internal calls